### DISCUSSION TITLE
Why I like and prefer xUnit

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#69] [General]
# Why I like and prefer xUnit

https://steven-giesel.com/blogPost/c24c5326-22f4-4035-9f87-a20e4814121d

**Comment by simonb65** [comment]
Thanks for the blog but ..... it is a tad naive.  Test isolation is very important. Many times I've had to undo test classes where devs had created a single SUT object for the whole class (what a bad pattern). Just create an instance of the object you need for each test. Relying on the individual fucntionality between the 2 platforms creates a dependency most devs wont appreciate. Just make the tests clear.

If there's complex construction, then use private static helper methods. These create mocks as required and return the test object. A new instance for each test.

I've used both xUnit and nUnit for years and now almost exclusively use nUnit now as control over test cases and labels is much better. When you have > 500 tests for a project, then being able to manage this is really important.

**Reply by linkdotnet** [reply]
Thanks, @simonb65, and yes, it is written a bit naive. Overall the framework is a means to an end and everyone should evaluate their specific needs. Test isolation is one of the major differences from the other "big players".

> nUnit now as control over test cases and labels is much better.

How does it differ from xUnit in this regard? xUnit also supports labels and categories.

**Comment by Melchy** [comment]
This was andvantage of xUnit for many years but Nunit added attribute  [assembly: FixtureLifeCycle(LifeCycle.InstancePerTestCase)] in version 3.13. This attribute forces Nunit to create class instance per test.

**Reply by linkdotnet** [reply]
Wow - I did not know that! Thanks for the info @Melchy !
